ksql-parser/src/main/java/io/confluent/ksql/parser/properties/with/CreateSourceProperties.java

```diff
@@ -125,4 +143,21 @@ public CreateSourceProperties withPartitionsAndReplicas(
 
     return new CreateSourceProperties(originals);
   }
+
+  private Optional<SerdeFactory<Windowed<String>>> getTimedWindowSerdeFactory() {
+    final Optional<String> windowSize = Optional.ofNullable(
+        getString(CreateConfigs.WINDOW_SIZE_PROPERTY))
+        .map(String::toUpperCase);
+    if (!windowSize.isPresent()) {
+      throw new KsqlException("Tumbling and Hopping window types should set "
+          + CreateConfigs.WINDOW_SIZE_PROPERTY + " in the WITH clause.");
+    }
+    final String[] sizeParts = StringUtil.cleanQuotes(windowSize.get()).split(" ");
+    final long size = Long.parseLong(sizeParts[0]);
```

Comment:
This duplicates code in `validateWindowSizeProperty`.  We should have one method (in this class) that can handle converting the string.  We should call than from the constructor to validate everything, and call it again in the getter to get the value to pass to the WindowedSerdes.